tests/integration/pilot/workloadentry_test.go

```diff
@@ -0,0 +1,157 @@
+//go:build integ
+// +build integ
+
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pilot
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/test/echo/common/scheme"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/check"
+	commonDeployment "istio.io/istio/pkg/test/framework/components/echo/common/deployment"
+	"istio.io/istio/pkg/test/framework/components/echo/common/ports"
+	"istio.io/istio/pkg/test/framework/components/echo/deployment"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/resource/config/apply"
+	"istio.io/istio/pkg/test/util/retry"
+)
+
+func TestWorkloadEntry(t *testing.T) {
+	// nolint: staticcheck
+	framework.NewTest(t).
+		RequiresMinClusters(2).
+		Features("traffic.reachability").
+		Run(func(t framework.TestContext) {
+			ist, err := istio.Get(t)
+			if err != nil {
+				t.Fatal(err)
+			}
+			clusterCfg := t.Clusters().Default()
+			namespaceName := apps.Namespace.Name()
+
+			// Create a echo deployment "z" in the echos namespace.
+			t.Logf("Deploy an echo instance in namespace %s on cluster %s", namespaceName, clusterCfg.Name())
+			deployment.New(t, clusterCfg).
```

Comment:
Do we need a new one? or can just use an existing one like `a`?

Reply:
My original implementation reused an existing deployment and (along the way I added) assertion that traffic reached the intended cluster. I found that it was more likely to flake when reusing an existing deployment. I tried a few different iterations of that but the results were never as stable as creating a new deployment.

Reply:
That seems like there may be a bug? I am not sure what difference a new vs old one would make in a properly functioning system?

Reply:
I didn't explain my reasoning very well here. The issue I was encountering is the EW gateway in the primary would also discover pod `a` in the remote cluster since it's a flat network. It was therefore possible that when hitting the EW gateway the requested ended up at remote `a` instead of primary `a` and my test would fail because it never reached the correct cluster. I could look at the envoy clusters statistics and see that successful requests went to the EW gateway's IP when the test failed.